Populate process ENV from EnvManager by default (#723)
wordpress-mobile/release-toolkit: CHANGELOG.md

### Breaking Changes

> - `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win) [#578].

Author: ```suggestion
- `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win) [#723].
```

> - `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win), so fastlane actions that read their `default_value:` from `ENV` find loaded secrets without callers having to thread them through explicitly. Pass `mutate_env: false` to opt out and keep the parse-only / instance-isolation semantics introduced in [#578]. `EnvManager.reset!` now also rolls back the keys it added.

Author: ```suggestion
- `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win) [#578].
```

> - `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win), so fastlane actions that read their `default_value:` from `ENV` find loaded secrets without callers having to thread them through explicitly. Pass `mutate_env: false` to opt out and keep the parse-only / instance-isolation semantics introduced in [#578]. `EnvManager.reset!` now also rolls back the keys it added. [#XXX]

Author: Already addressed in the current PR head: the changelog entry now references `#723` instead of the placeholder.

*Posted by Codex (GPT-5) on behalf of @mokagio with approval.*

> - `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win), so fastlane actions that read their `default_value:` from `ENV` find loaded secrets without callers having to thread them through explicitly. Pass `mutate_env: false` to opt out and keep the parse-only / instance-isolation semantics introduced in [#578]. `EnvManager.reset!` now also rolls back the keys it added. [#XXX]

Author: Already addressed in the current PR head: the changelog entry now references `#723` instead of the placeholder.

*Posted by Codex (GPT-5) on behalf of @mokagio with approval.*

### Breaking Changes

- The `ios_build_preflight` and `android_build_preflight` actions have been removed. [#773]

> - `EnvManager`: populate the process `ENV` from the loaded `.env` file by default (no-override semantics — pre-existing `ENV` values win) [#723].

Author: ```suggestion
- `EnvManager` now populates the process `ENV` from the loaded `.env` file by default [#723].
```
